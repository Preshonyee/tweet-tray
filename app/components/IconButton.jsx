import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';

const IconButtonStyle = Styled.button`
    -webkit-app-region: no-drag;
    user-select: none;
    width: auto;
    height: 100%;
    border: 0px;
    outline: 0px;
    background-color: transparent;
    transition: all 0.3s ease 0s;

    &:disabled, &[disabled], &:hover {
        opacity: 0.75;
        cursor: pointer !important;
        transition: all 0.3s ease 0s;
    }

    &:disabled, &[disabled] {
      cursor: default !important;
    }
`;

const ImageStyle = Styled.img`
    display: block;
    pointer-events: none;
    margin-top: 1px;
    width: 25px;
    height: 25px;

    &:hover {
      cursor: pointer !important;
    }
`;

const IconButton = (props) => {
  const {
    disabled, iconSrc, altText, onClick,
  } = props;

  return (
    <IconButtonStyle
      onClick={onClick}
      disabled={disabled}
    >
      <ImageStyle
        src={iconSrc}
        alt={altText}
      />
    </IconButtonStyle>
  );
};

IconButton.propTypes = {
  disabled: PropTypes.bool,
  altText: PropTypes.string.isRequired,
  iconSrc: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

IconButton.defaultProps = {
  disabled: false,
};
export default IconButton;
