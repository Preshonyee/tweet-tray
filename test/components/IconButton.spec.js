import React from 'react';
import Enzyme, { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import * as testData from '../testData';

import IconButton from '../../app/components/IconButton';

Enzyme.configure({ adapter: new Adapter(), });

const mockCallBack = jest.fn();

function setup() {
  const component = shallow(<IconButton
    color={testData.profileLinkColor}
    icon="settings"
    onClick={mockCallBack}
  />);
  return {
    component,
    image: component.find('IconButton__ImageStyle'),
  };
}

describe('IconButton component', () => {
  it('should render button', () => {
    const { component, } = setup();
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render image inside button', () => {
    const { image, } = setup();
    expect(shallowToJson(image)).toMatchSnapshot();
  });
});
