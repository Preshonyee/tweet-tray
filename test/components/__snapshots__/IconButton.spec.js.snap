// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component should render button 1`] = `
<IconButton__IconButtonStyle
  disabled={false}
  onClick={[MockFunction]}
>
  <IconButton__ImageContainerStyle>
    <Icon
      color="#1da1f2"
      name="settings"
      size={27}
      theme="day"
    />
  </IconButton__ImageContainerStyle>
</IconButton__IconButtonStyle>
`;

exports[`IconButton component should render image inside button 1`] = `null`;
